### ISSUE TITLE
The backend api couldn't be open in remote machine

### ISSUE THREAD

**Issue by wangyibin** [lidaof/eg-react#132]
Hi @lidaof 
I have a problem with setup my own backend API in my remote host.
I install the eg-react follow with the steps of:
```
docker run -p 3000:3000 -p 3001:3001 --name epgg -it epgg/eg-react /bin/bash
cd /eg-react/frontend 
npm install
cd /eg-react/backend
npm install
npm run setup
npm start
cd /eg-react/frontend
npm start
```
And then I can use curl to connect backend successful in the docker container. But failed to open the backend API in my window computer. 
![image](https://user-images.githubusercontent.com/37402369/66373481-541ff180-e9db-11e9-8a21-b4d2812e98fd.png)
![image](https://user-images.githubusercontent.com/37402369/66373620-9e08d780-e9db-11e9-8e40-9ee0c3a62742.png)

Therefore, the frontend failed to fetch data.
![image](https://user-images.githubusercontent.com/37402369/66373832-24251e00-e9dc-11e9-806e-d4b9bcf899f4.png)

And I also attempt to use the proxy method which edits the package.json in the backend to solve this problem. But failed.
![image](https://user-images.githubusercontent.com/37402369/66373946-7cf4b680-e9dc-11e9-9c23-8aa8441307a6.png)

How can I solve this problem? Thank you for your reply.

YibinWang

**Comment by lidaof** [comment]
Hi @wangyibin , have you changed the `host` in backend's `main.js`? thanks for using the browser!

**Comment by wangyibin** [comment]
yes，I have tried to change the host in backend's main.js. But failed to start the backend.
![image](https://user-images.githubusercontent.com/37402369/66443892-867d2d80-ea73-11e9-8e56-652a0a576d14.png)
![image](https://user-images.githubusercontent.com/37402369/66443907-9ac12a80-ea73-11e9-9270-49b4dbd84ef1.png)
Thanks for your reply.

**Comment by lidaof** [comment]
you are welcome. is the IP address belongs to your server? what's the output of the `ifconfig` command?

**Comment by wangyibin** [comment]
yes, this is the output of my `ifconfig` command.
![image](https://user-images.githubusercontent.com/37402369/66454366-41b8bd00-ea9a-11e9-8224-161b11222b03.png)
I can open the frontend by this IP `http://210.34.81.56:3000`.
![image](https://user-images.githubusercontent.com/37402369/66454437-775da600-ea9a-11e9-9afd-06e6477b5010.png)

**Comment by lidaof** [comment]
Hi @wangyibin this is strange....not quite sure. My gut feeling is docker image cannot assign the host IP address....can you just try using localhost? in the package.json proxy, also use localhost.

**Comment by wangyibin** [comment]
I change the host address to `localhost` in `main.js` and `package.json`.
![image](https://user-images.githubusercontent.com/37402369/66530976-a6c4ef00-eb3c-11e9-8819-ec99204ad660.png)
![image](https://user-images.githubusercontent.com/37402369/66530985-afb5c080-eb3c-11e9-8dfb-80b4f6e26736.png)
the backend can be run.
![image](https://user-images.githubusercontent.com/37402369/66531171-79c50c00-eb3d-11e9-9a26-80172a58619a.png)
 and then use the curl to test in the docker it also is ok.
![image](https://user-images.githubusercontent.com/37402369/66531437-5a7aae80-eb3e-11e9-9311-063e32ebf51c.png)

and change the `AWS_API='http://localhost:3001`, but it failed to connect to the backend when open the frontend.
![image](https://user-images.githubusercontent.com/37402369/66531052-fdcac400-eb3c-11e9-8e58-92e002d2a230.png)

There is a message when I run `npm install` in the backend, is this affect?
![image](https://user-images.githubusercontent.com/37402369/66531224-a0834280-eb3d-11e9-8241-60287dc715c9.png)
![image](https://user-images.githubusercontent.com/37402369/66531252-ad079b00-eb3d-11e9-9c06-a7fbbfd41dca.png)

**Comment by lidaof** [comment]
one more try, can you change the AWS_API to an empty string?

**Comment by lidaof** [comment]
ok, sorry about that @wangyibin . I will need do some tests on the docker image. by the way, is it possible you can run the mirror directly with node.js (without docker)?

**Comment by wangyibin** [comment]
Hi @lidaof 
Sorry for the late reply. I have solved this problem when I use docker to set up my own API.
I used docker mapping a port `3003`  to `3000`.  So I add the `"proxy": "http://210.34.81.56:3003" `to` the `backend/package.json`
 
![image](https://user-images.githubusercontent.com/37402369/66727211-34bc1500-ee70-11e9-8e86-bd5da41cf80f.png)
And then empty the `AWS_API`
And add the `"proxy": "http://localhost:3001" to the `frontend/package.json`.
Finally, the browser works. 
![image](https://user-images.githubusercontent.com/37402369/66727258-8d8bad80-ee70-11e9-9152-72707fa75f0a.png)

I don't know if my steps are redundant. But the browser works. Thank you for your help.
